config: add info about output.scriptType (#3818)
webpack/webpack.js.org: src/content/configuration/output.md

T> Read the [authoring libraries guide](/guides/author-libraries/) guide for more information on `output.library` as well as `output.libraryTarget`.

## ouput.scriptType

> `string='module'` `bool=false`

Reviewer: ```suggestion
`string: 'module' | 'text/javascript'` `boolean = false`
```

Reviewer: Please pay attention to defaults, this option only takes two possible strings, need to list them, also formatting: https://webpack.js.org/contribute/writers-guide/#configuration-defaults-and-types

> This option enables loading async chunks via a custom script type, such as script type=`module`.

Reviewer: ```suggestion
This option allows loading asynchronous chunks with a custom script type, such as `<script type="module" ...>`.
```

> T> If `output.module` is `true` then by deafult `scriptType` would be set to `module` and if it is `false` then it would be set to `false`.

Reviewer: ```suggestion
T> If [`output.module`](#outputmodule) is set to `true`, `ouput.scriptType` will default to `'module'` instead of `false`.
```
